### PR TITLE
Add an Executor thread and a Deleter.

### PR DESCRIPTION
**Pull request by nomis52**
This provides async deletion of pointers.

### CODE REVIEW

File: common/thread/Makefile.mk [hunk 1]
```diff
@@ -10,7 +11,9 @@ common_libolacommon_la_SOURCES += \
 
 # TESTS
 ##################################################
-test_programs += common/thread/ThreadTester common/thread/FutureTester
+test_programs += common/thread/ExectutorThreadTester \
```
Comment (peternewman) [inline common/thread/Makefile.mk]: SPaG Executor

File: common/thread/Makefile.mk [hunk 2]
```diff
@@ -21,3 +24,8 @@ common_thread_ThreadTester_LDADD = $(COMMON_TESTING_LIBS)
 common_thread_FutureTester_SOURCES = common/thread/FutureTest.cpp
 common_thread_FutureTester_CXXFLAGS = $(COMMON_TESTING_FLAGS)
 common_thread_FutureTester_LDADD = $(COMMON_TESTING_LIBS)
+
+common_thread_ExectutorThreadTester_SOURCES = \
```
Comment (peternewman) [inline common/thread/Makefile.mk]: SPaG Executor

File: include/ola/thread/ExecutorThread.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   * @returns true if the thread started, false if the thread was already
+   *   running.
+   *
+   * Not thread safe.
+   */
+  bool Start();
+
+  /**
+   * @brief Stop this Executor thread.
+   * @returns true if the Executor was stopped, false if it wasn't running.
```
Comment (peternewman) [inline include/ola/thread/ExecutorThread.h]: What happens if the executor failed to stop?

File: include/ola/util/Deleter.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ * @tparam T any type.
+ * @param t The pointer to delete. Ownership is transferred.
+ */
+template <typename T>
+void Deleter(T* t) {
+  delete t;
+}
+
+/**
+ * @brief Create a callack that deletes the object.
```
Comment (peternewman) [inline include/ola/util/Deleter.h]: SPaG callback

**Comment by peternewman** [comment]
LGTM, just minor comments.

File: common/thread/Makefile.mk (hunk 1 above)
Comment (nomis52) [inline common/thread/Makefile.mk]: Done

File: common/thread/Makefile.mk (hunk 2 above)
Comment (nomis52) [inline common/thread/Makefile.mk]: Done

File: include/ola/util/Deleter.h (hunk 4 above)
Comment (nomis52) [inline include/ola/util/Deleter.h]: Fixed.

File: include/ola/thread/ExecutorThread.h (hunk 3 above)
Comment (nomis52) [inline include/ola/thread/ExecutorThread.h]: In all likelihood the program deadlocks.  As far as I can see the only way to trigger that is to call Stop() simultaneously from two different threads. I added a comment.

**Pull request merged**